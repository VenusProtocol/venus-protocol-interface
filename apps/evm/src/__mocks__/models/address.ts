const address = '0x3d759121234cd36F8124C21aFe1c6852d2bEd848';

export const altAddress = '0xa258a693A403b7e98fd05EE9e1558C760308cFC7';

export default address;
