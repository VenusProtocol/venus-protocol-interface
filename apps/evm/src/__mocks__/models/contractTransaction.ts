import { ChainId } from 'types';
import type { Transaction } from 'viem';

const contractTransaction: Transaction = {
  hash: '0xfake-transaction-hash',
  type: 'legacy',
  from: '0x1d759121234cd36F8124C21aFe1c6852d2bEd848',
  to: '0x2d759121234cd36F8124C21aFe1c6852d2bEd834',
  nonce: 10,
  value: 0n,
  chainId: ChainId.BSC_TESTNET,
  blockHash: '0xfake-block-hash',
  blockNumber: 1123213n,
  gas: 12n,
  gasPrice: 1n,
  input: '0xfake-input',
  r: '0xfake-r',
  s: '0xfake-s',
  v: 3n,
  transactionIndex: 1,
  typeHex: '0xfake-type-hex',
};

export default contractTransaction;
