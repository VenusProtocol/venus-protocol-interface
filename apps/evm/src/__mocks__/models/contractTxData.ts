import type { ContractTxData } from 'types';

const contractTxData: ContractTxData<any, any> = {
  contract: {},
  methodName: 'fakeMethod',
  args: [],
};

export default contractTxData;
