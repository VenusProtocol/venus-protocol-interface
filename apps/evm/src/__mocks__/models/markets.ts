import BigNumber from 'bignumber.js';

import { Market } from 'types';

export const markets: Market[] = [
  {
    address: '0x74469281310195a04840daf6edf576f559a3de80',
    borrowerCount: 306,
    supplierCount: 898,
    totalXvsDistributedTokens: new BigNumber('318096'),
  },
  {
    address: '0xd5c4c2e2facbeb59d0216d0595d63fcdc6f9a1a7',
    borrowerCount: 232,
    supplierCount: 638,
    totalXvsDistributedTokens: new BigNumber('261028'),
  },
  {
    address: '0xb7526572ffe56ab9d7489838bf2e18e3323b441a',
    borrowerCount: 315,
    supplierCount: 736,
    totalXvsDistributedTokens: new BigNumber('211354'),
  },
  {
    address: '0x08e0a5575de71037ae36abfafb516595fe68e5e4',
    borrowerCount: 533,
    supplierCount: 2235,
    totalXvsDistributedTokens: new BigNumber('928041'),
  },
  {
    address: '0x2e7222e51c0f6e98610a1543aa3836e092cde62c',
    borrowerCount: 261,
    supplierCount: 1700,
    totalXvsDistributedTokens: new BigNumber('924581'),
  },
  {
    address: '0x6d6f697e34145bb95c54e77482d97cc261dc237e',
    borrowerCount: 77,
    supplierCount: 326,
    totalXvsDistributedTokens: new BigNumber('14741'),
  },
  {
    address: '0xb6e9322c49fd75a367fcb17b0fcd62c5070ebcbe',
    borrowerCount: 51,
    supplierCount: 415,
    totalXvsDistributedTokens: new BigNumber('269883'),
  },
  {
    address: '0x162d005f0fff510e54958cfc5cf32a3180a84aab',
    borrowerCount: 51,
    supplierCount: 211,
    totalXvsDistributedTokens: new BigNumber('39187'),
  },
  {
    address: '0xafc13bc065abee838540823431055d2ea52eba52',
    borrowerCount: 15,
    supplierCount: 150,
    totalXvsDistributedTokens: new BigNumber('56938'),
  },
  {
    address: '0x488ab2826a154da01cc4cc16a8c83d4720d3ca2c',
    borrowerCount: 31,
    supplierCount: 188,
    totalXvsDistributedTokens: new BigNumber('54890'),
  },
  {
    address: '0x37c28de42ba3d22217995d146fc684b2326ede64',
    borrowerCount: 19,
    supplierCount: 157,
    totalXvsDistributedTokens: new BigNumber('15830'),
  },
  {
    address: '0xf912d3001caf6dc4add366a62cc9115b4303c9a9',
    borrowerCount: 20,
    supplierCount: 9,
    totalXvsDistributedTokens: new BigNumber('2487'),
  },
  {
    address: '0x3619bddc61189f33365cc572df3a68fb3b316516',
    borrowerCount: 7,
    supplierCount: 152,
    totalXvsDistributedTokens: new BigNumber('11'),
  },
  {
    address: '0xedac03d29ff74b5fdc0cc936f6288312e1459bc6',
    borrowerCount: 9,
    supplierCount: 195,
    totalXvsDistributedTokens: new BigNumber('4473'),
  },
  {
    address: '0x714db6c38a17883964b68a07d56ce331501d9eb6',
    borrowerCount: 6,
    supplierCount: 144,
    totalXvsDistributedTokens: new BigNumber('805'),
  },
  {
    address: '0x3a00d9b02781f47d033bad62edc55fbf8d083fb0',
    borrowerCount: 5,
    supplierCount: 138,
    totalXvsDistributedTokens: new BigNumber('3480'),
  },
  {
    address: '0x369fea97f6fb7510755dca389088d9e2e2819278',
    borrowerCount: 3,
    supplierCount: 133,
    totalXvsDistributedTokens: new BigNumber('4020'),
  },
  {
    address: '0xf206af85bc2761c4f876d27bd474681cfb335efa',
    borrowerCount: 2,
    supplierCount: 5,
    totalXvsDistributedTokens: new BigNumber('601'),
  },
  {
    address: '0x9c3015191d39cf1930f92eb7e7bcbd020bca286a',
    borrowerCount: 0,
    supplierCount: 3,
    totalXvsDistributedTokens: new BigNumber('473'),
  },
];
