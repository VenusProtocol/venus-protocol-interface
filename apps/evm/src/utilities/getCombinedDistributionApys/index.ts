import BigNumber from 'bignumber.js';

import type { Asset, TokenDistribution } from 'types';

interface AggregatePercentagesInput {
  distributions: TokenDistribution[];
}

const aggregatePercentages = ({ distributions }: AggregatePercentagesInput) =>
  distributions.reduce<{
    apyRewardsPercentage: BigNumber;
    apyPrimePercentage: BigNumber;
    apyPrimeSimulationPercentage: BigNumber;
  }>(
    (acc, distribution) => {
      if (distribution.type === 'venus' || distribution.type === 'merkl') {
        return {
          ...acc,
          apyRewardsPercentage: acc.apyRewardsPercentage.plus(distribution.apyPercentage),
        };
      }

      if (distribution.type === 'prime') {
        return {
          ...acc,
          apyPrimePercentage: acc.apyPrimePercentage.plus(distribution.apyPercentage),
        };
      }

      if (distribution.type === 'primeSimulation') {
        return {
          ...acc,
          apyPrimeSimulationPercentage: acc.apyPrimeSimulationPercentage.plus(
            distribution.apyPercentage,
          ),
        };
      }

      return acc;
    },
    {
      apyRewardsPercentage: new BigNumber(0),
      apyPrimePercentage: new BigNumber(0),
      apyPrimeSimulationPercentage: new BigNumber(0),
    },
  );

export interface GetCombinedDistributionApysInput {
  asset: Asset;
}

const getCombinedDistributionApys = ({ asset }: GetCombinedDistributionApysInput) => {
  const supply = aggregatePercentages({ distributions: asset.supplyTokenDistributions });
  const borrow = aggregatePercentages({ distributions: asset.borrowTokenDistributions });

  const totalSupplyApyBoostPercentage = supply.apyRewardsPercentage.plus(supply.apyPrimePercentage);
  const totalBorrowApyBoostPercentage = borrow.apyRewardsPercentage.plus(borrow.apyPrimePercentage);

  return {
    supplyApyRewardsPercentage: supply.apyRewardsPercentage,
    borrowApyRewardsPercentage: borrow.apyRewardsPercentage,
    supplyApyPrimePercentage: supply.apyPrimePercentage,
    borrowApyPrimePercentage: borrow.apyPrimePercentage,
    supplyApyPrimeSimulationPercentage: supply.apyPrimeSimulationPercentage,
    borrowApyPrimeSimulationPercentage: borrow.apyPrimeSimulationPercentage,
    totalSupplyApyBoostPercentage,
    totalBorrowApyBoostPercentage,
    totalSupplyApyPercentage: asset.supplyApyPercentage.plus(totalSupplyApyBoostPercentage),
    totalBorrowApyPercentage: asset.borrowApyPercentage.minus(totalBorrowApyBoostPercentage),
  };
};

export default getCombinedDistributionApys;
