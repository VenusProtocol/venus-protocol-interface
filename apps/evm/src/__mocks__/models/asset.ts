import BigNumber from 'bignumber.js';

import { busd, usdc, usdt, xvs } from '__mocks__/models/tokens';
import { vBusd, vUsdc, vUsdt, vXvs } from '__mocks__/models/vTokens';

import { Asset } from 'types';

export const assetData: Asset[] = [
  {
    vToken: vXvs,
    supplyApyPercentage: new BigNumber('0.05225450324405023'),
    borrowApyPercentage: new BigNumber('-2.3062487835658776'),
    collateralFactor: 0.5,
    reserveFactor: 0.25,
    reserveTokens: new BigNumber(1000),
    cashTokens: new BigNumber(10),
    tokenPriceCents: new BigNumber('127.86734'),
    liquidityCents: new BigNumber(8036465875),
    supplierCount: 100,
    borrowerCount: 10,
    exchangeRateVTokens: new BigNumber(49.589181233),
    userWalletBalanceTokens: new BigNumber('100'),
    userWalletBalanceCents: new BigNumber('12786'),
    userSupplyBalanceTokens: new BigNumber('90'),
    userSupplyBalanceCents: new BigNumber('11508'),
    userBorrowBalanceTokens: new BigNumber('0'),
    userBorrowBalanceCents: new BigNumber('0'),
    isCollateralOfUser: true,
    userPercentOfLimit: 0,
    borrowBalanceCents: new BigNumber(70925716),
    supplyBalanceCents: new BigNumber(278311516),
    supplyBalanceTokens: new BigNumber('19339683254955736'),
    borrowBalanceTokens: new BigNumber('1852935.597521220541385584'),
    supplyPercentageRatePerBlock: new BigNumber('0.000001'),
    borrowPercentageRatePerBlock: new BigNumber('0.0000001'),
    supplyDistributions: [
      {
        type: 'rewardDistributor',
        token: xvs,
        apyPercentage: new BigNumber('0.11720675342484096'),
        dailyDistributedTokens: new BigNumber('9999999.5'),
      },
    ],
    borrowDistributions: [
      {
        type: 'rewardDistributor',
        token: xvs,
        apyPercentage: new BigNumber('4.17469243006608279'),
        dailyDistributedTokens: new BigNumber('9999999.5'),
      },
    ],
  },
  {
    vToken: vUsdc,
    supplyApyPercentage: new BigNumber('3.887242555711379188'),
    borrowApyPercentage: new BigNumber('-5.361233028654066'),
    collateralFactor: 0.8,
    reserveFactor: 0.2,
    reserveTokens: new BigNumber(1000),
    cashTokens: new BigNumber(10),
    tokenPriceCents: new BigNumber('99.99364'),
    liquidityCents: new BigNumber(1702951959),
    supplierCount: 100,
    borrowerCount: 10,
    exchangeRateVTokens: new BigNumber(1),
    userWalletBalanceTokens: new BigNumber('0'),
    userWalletBalanceCents: new BigNumber('0'),
    userSupplyBalanceTokens: new BigNumber('100'),
    userSupplyBalanceCents: new BigNumber('9999'),
    userBorrowBalanceTokens: new BigNumber('0'),
    userBorrowBalanceCents: new BigNumber('0'),
    isCollateralOfUser: false,
    userPercentOfLimit: 0,
    borrowBalanceCents: new BigNumber(858721657509436),
    supplyBalanceCents: new BigNumber(1000183891879506),
    supplyBalanceTokens: new BigNumber('47171999131879271527200010830'),
    borrowBalanceTokens: new BigNumber('73128318.509651061457900627'),
    supplyPercentageRatePerBlock: new BigNumber('0.000002'),
    borrowPercentageRatePerBlock: new BigNumber('0.0000002'),
    supplyDistributions: [
      {
        type: 'rewardDistributor',
        token: xvs,
        apyPercentage: new BigNumber('1.353105649796123742'),
        dailyDistributedTokens: new BigNumber('9999999.5'),
      },
      {
        type: 'primeSimulation',
        token: usdc,
        apyPercentage: new BigNumber('0.953105649796123742'),
        referenceValues: {
          userBorrowBalanceTokens: new BigNumber('1000'),
          userSupplyBalanceTokens: new BigNumber('1000'),
          userXvsStakedTokens: new BigNumber('1000'),
        },
      },
      {
        type: 'prime',
        token: usdc,
        apyPercentage: new BigNumber('0.753105649796123742'),
      },
    ],
    borrowDistributions: [
      {
        type: 'rewardDistributor',
        token: xvs,
        apyPercentage: new BigNumber('1.670327607690572731'),
        dailyDistributedTokens: new BigNumber('9999999.5'),
      },
      {
        type: 'primeSimulation',
        token: usdc,
        apyPercentage: new BigNumber('1.233105649796123742'),
        referenceValues: {
          userBorrowBalanceTokens: new BigNumber('1000'),
          userSupplyBalanceTokens: new BigNumber('1000'),
          userXvsStakedTokens: new BigNumber('1000'),
        },
      },
      {
        type: 'prime',
        token: usdc,
        apyPercentage: new BigNumber('0.913105649796123742'),
      },
    ],
  },
  {
    vToken: vUsdt,
    supplyApyPercentage: new BigNumber('3.593608909332766999'),
    borrowApyPercentage: new BigNumber('-4.9748661428011145'),
    collateralFactor: 0.8,
    reserveFactor: 0.2,
    reserveTokens: new BigNumber(1000),
    cashTokens: new BigNumber(10),
    tokenPriceCents: new BigNumber('100.024602'),
    liquidityCents: new BigNumber(5534102886),
    supplierCount: 100,
    borrowerCount: 10,
    exchangeRateVTokens: new BigNumber(0.981982),
    userWalletBalanceTokens: new BigNumber('900'),
    userWalletBalanceCents: new BigNumber('90000'),
    userSupplyBalanceTokens: new BigNumber('0'),
    userSupplyBalanceCents: new BigNumber('0'),
    userBorrowBalanceTokens: new BigNumber('40'),
    userBorrowBalanceCents: new BigNumber('4000'),
    isCollateralOfUser: false,
    userPercentOfLimit: 0,
    borrowBalanceCents: new BigNumber(3158444721),
    supplyBalanceCents: new BigNumber(1098041201011568),
    supplyBalanceTokens: new BigNumber('5029972090817266864401527367893625'),
    borrowBalanceTokens: new BigNumber('232511166.920938849475104194'),
    supplyPercentageRatePerBlock: new BigNumber('0.000003'),
    borrowPercentageRatePerBlock: new BigNumber('0.0000003'),
    supplyDistributions: [
      {
        type: 'rewardDistributor',
        token: xvs,
        apyPercentage: new BigNumber('0.421719501189155143'),
        dailyDistributedTokens: new BigNumber('9999999.5'),
      },
      {
        type: 'primeSimulation',
        token: usdt,
        apyPercentage: new BigNumber('1.753105649796123742'),
        referenceValues: {
          userBorrowBalanceTokens: new BigNumber('1000'),
          userSupplyBalanceTokens: new BigNumber('1000'),
          userXvsStakedTokens: new BigNumber('1000'),
        },
      },
    ],
    borrowDistributions: [
      {
        type: 'rewardDistributor',
        token: xvs,
        apyPercentage: new BigNumber('0.522209972682294832'),
        dailyDistributedTokens: new BigNumber('9999999.5'),
      },
      {
        type: 'primeSimulation',
        token: usdt,
        apyPercentage: new BigNumber('1.015310564979612374'),
        referenceValues: {
          userBorrowBalanceTokens: new BigNumber('1000'),
          userSupplyBalanceTokens: new BigNumber('1000'),
          userXvsStakedTokens: new BigNumber('1000'),
        },
      },
    ],
  },
  {
    vToken: vBusd,
    supplyApyPercentage: new BigNumber('2.886396363044176106'),
    borrowApyPercentage: new BigNumber('-4.050271277344538'),
    collateralFactor: 0.8,
    reserveFactor: 0.2,
    reserveTokens: new BigNumber(1000),
    cashTokens: new BigNumber(10),
    tokenPriceCents: new BigNumber('100.000922'),
    liquidityCents: new BigNumber(3654492935),
    supplierCount: 100,
    borrowerCount: 10,
    exchangeRateVTokens: new BigNumber(1.000003),
    userWalletBalanceTokens: new BigNumber('110'),
    userWalletBalanceCents: new BigNumber('11000'),
    userSupplyBalanceTokens: new BigNumber('0'),
    userSupplyBalanceCents: new BigNumber('0'),
    userBorrowBalanceTokens: new BigNumber('50'),
    userBorrowBalanceCents: new BigNumber('5000'),
    isCollateralOfUser: false,
    userPercentOfLimit: 0,
    borrowBalanceCents: new BigNumber(83910350502),
    supplyBalanceCents: new BigNumber(1054707853878),
    supplyBalanceTokens: new BigNumber('51881081291203672464'),
    borrowBalanceTokens: new BigNumber('142662020.229587308931217432'),
    supplyPercentageRatePerBlock: new BigNumber('0.000004'),
    borrowPercentageRatePerBlock: new BigNumber('0.0000004'),
    supplyDistributions: [
      {
        type: 'rewardDistributor',
        token: xvs,
        apyPercentage: new BigNumber('0.678420831753642169'),
        dailyDistributedTokens: new BigNumber('9999999.5'),
      },
      {
        type: 'primeSimulation',
        token: busd,
        apyPercentage: new BigNumber('1.753105649796123742'),
        referenceValues: {
          userBorrowBalanceTokens: new BigNumber('1000'),
          userSupplyBalanceTokens: new BigNumber('1000'),
          userXvsStakedTokens: new BigNumber('1000'),
        },
      },
      {
        type: 'prime',
        token: xvs,
        apyPercentage: new BigNumber(0),
      },
    ],
    borrowDistributions: [
      {
        type: 'rewardDistributor',
        token: xvs,
        apyPercentage: new BigNumber('0.852697602175970714'),
        dailyDistributedTokens: new BigNumber('9999999.5'),
      },
      {
        type: 'primeSimulation',
        token: busd,
        apyPercentage: new BigNumber('1.753105649796123742'),
        referenceValues: {
          userBorrowBalanceTokens: new BigNumber('1000'),
          userSupplyBalanceTokens: new BigNumber('1000'),
          userXvsStakedTokens: new BigNumber('1000'),
        },
      },
      {
        type: 'prime',
        token: busd,
        apyPercentage: new BigNumber('0.913105649796123742'),
      },
    ],
  },
];
