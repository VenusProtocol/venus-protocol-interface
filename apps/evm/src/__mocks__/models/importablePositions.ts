import { busd, usdc, usdt, weth, xvs } from '__mocks__/models/tokens';
import type { ImportableProtocol, ImportableSupplyPosition } from 'types';

export const importablePositions: {
  [protocol in ImportableProtocol]: ImportableSupplyPosition[];
} = {
  aave: [
    {
      aTokenAddress: `0xfakeA${xvs.symbol}Address`,
      protocol: 'aave',
      supplyApyPercentage: 99.99,
      tokenAddress: xvs.address,
      userATokenBalanceMantissa: 20000000000000000000n,
      userATokenBalanceWithInterestsMantissa: 24000000000000000000n,
      userSupplyBalanceMantissa: 40000000000000000000n,
    },
    {
      aTokenAddress: `0xfakeA${usdc.symbol}Address`,
      protocol: 'aave',
      supplyApyPercentage: 0.01,
      tokenAddress: usdc.address,
      userATokenBalanceMantissa: 10000000n,
      userATokenBalanceWithInterestsMantissa: 12000000n,
      userSupplyBalanceMantissa: 50000000n,
    },
    {
      aTokenAddress: `0xfakeA${usdt.symbol}Address`,
      protocol: 'aave',
      supplyApyPercentage: 0.15,
      tokenAddress: usdt.address,
      userATokenBalanceMantissa: 10000000n,
      userATokenBalanceWithInterestsMantissa: 12000000n,
      userSupplyBalanceMantissa: 50000000n,
    },
    {
      aTokenAddress: `0xfakeA${busd.symbol}Address`,
      protocol: 'aave',
      supplyApyPercentage: 0.02,
      tokenAddress: busd.address,
      userATokenBalanceMantissa: 1n,
      userATokenBalanceWithInterestsMantissa: 1n,
      userSupplyBalanceMantissa: 1n,
    },
    {
      aTokenAddress: `0xfakeA${weth.symbol}Address`,
      protocol: 'aave',
      supplyApyPercentage: 0.09,
      tokenAddress: weth.address,
      userATokenBalanceMantissa: 180000000000000000000n,
      userATokenBalanceWithInterestsMantissa: 186000000000000000000n,
      userSupplyBalanceMantissa: 200000000000000000000n,
    },
  ],
};
