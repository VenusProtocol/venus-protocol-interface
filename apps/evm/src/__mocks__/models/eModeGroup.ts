import type { Asset, EModeGroup } from 'types';
import { assetData } from './asset';

const generateEModeGroup = ({
  groupAssets,
  id,
  name,
}: { groupAssets: Asset[]; id: number; name: string }): EModeGroup => ({
  id,
  name,
  assetSettings: groupAssets.map(a => ({
    vToken: a.vToken,
    collateralFactor: a.userCollateralFactor + 0.1,
    liquidationThresholdPercentage: a.userLiquidationThresholdPercentage + 12,
    liquidationPenaltyPercentage: a.userLiquidationThresholdPercentage - 50,
    liquidityCents: a.liquidityCents.toNumber(),
    liquidityTokens: a.cashTokens,
    isBorrowable: a.isBorrowable,
  })),
});

export const eModeGroups: EModeGroup[] = [
  generateEModeGroup({
    id: 0,
    name: 'Stablecoins',
    groupAssets: assetData.slice(0, 3),
  }),
  generateEModeGroup({
    id: 1,
    name: 'DeFi',
    groupAssets: assetData.slice(2, 4),
  }),
  generateEModeGroup({
    id: 2,
    name: '#ToTheMoon',
    groupAssets: assetData.slice(5, 8),
  }),
];
