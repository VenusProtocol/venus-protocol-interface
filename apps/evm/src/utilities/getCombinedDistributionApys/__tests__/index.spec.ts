import { assetData as assets } from '__mocks__/models/asset';

import getCombinedDistributionApys from '..';

describe('utilities/getCombinedDistributionApys', () => {
  it('calculates combined distribution APYS correctly', () => {
    assets.forEach(asset => {
      const result = getCombinedDistributionApys({
        asset,
      });

      expect(result).toMatchSnapshot();
    });
  });
});
