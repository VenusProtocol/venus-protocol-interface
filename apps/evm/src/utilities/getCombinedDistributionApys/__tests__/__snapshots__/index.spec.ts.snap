// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`utilities/getCombinedDistributionApys > calculates combined distribution APYS correctly 1`] = `
{
  "borrowApyPrimePercentage": "0",
  "borrowApyPrimeSimulationPercentage": "0",
  "borrowApyRewardsPercentage": "4.17469243006608279",
  "supplyApyPrimePercentage": "0",
  "supplyApyPrimeSimulationPercentage": "0",
  "supplyApyRewardsPercentage": "0.11720675342484096",
  "totalBorrowApyBoostPercentage": "4.17469243006608279",
  "totalBorrowApyPercentage": "-6.48094121363196039",
  "totalSupplyApyBoostPercentage": "0.11720675342484096",
  "totalSupplyApyPercentage": "0.16946125666889119",
}
`;

exports[`utilities/getCombinedDistributionApys > calculates combined distribution APYS correctly 2`] = `
{
  "borrowApyPrimePercentage": "0.913105649796123742",
  "borrowApyPrimeSimulationPercentage": "1.233105649796123742",
  "borrowApyRewardsPercentage": "1.670327607690572731",
  "supplyApyPrimePercentage": "0.753105649796123742",
  "supplyApyPrimeSimulationPercentage": "0.953105649796123742",
  "supplyApyRewardsPercentage": "1.353105649796123742",
  "totalBorrowApyBoostPercentage": "2.583433257486696473",
  "totalBorrowApyPercentage": "-7.944666286140762473",
  "totalSupplyApyBoostPercentage": "2.106211299592247484",
  "totalSupplyApyPercentage": "5.993453855303626672",
}
`;

exports[`utilities/getCombinedDistributionApys > calculates combined distribution APYS correctly 3`] = `
{
  "borrowApyPrimePercentage": "0",
  "borrowApyPrimeSimulationPercentage": "1.015310564979612374",
  "borrowApyRewardsPercentage": "0.522209972682294832",
  "supplyApyPrimePercentage": "0",
  "supplyApyPrimeSimulationPercentage": "1.753105649796123742",
  "supplyApyRewardsPercentage": "0.421719501189155143",
  "totalBorrowApyBoostPercentage": "0.522209972682294832",
  "totalBorrowApyPercentage": "-5.497076115483409332",
  "totalSupplyApyBoostPercentage": "0.421719501189155143",
  "totalSupplyApyPercentage": "4.015328410521922142",
}
`;

exports[`utilities/getCombinedDistributionApys > calculates combined distribution APYS correctly 4`] = `
{
  "borrowApyPrimePercentage": "0.913105649796123742",
  "borrowApyPrimeSimulationPercentage": "1.753105649796123742",
  "borrowApyRewardsPercentage": "0.852697602175970714",
  "supplyApyPrimePercentage": "0",
  "supplyApyPrimeSimulationPercentage": "1.753105649796123742",
  "supplyApyRewardsPercentage": "0.678420831753642169",
  "totalBorrowApyBoostPercentage": "1.765803251972094456",
  "totalBorrowApyPercentage": "-5.816074529316632456",
  "totalSupplyApyBoostPercentage": "0.678420831753642169",
  "totalSupplyApyPercentage": "3.564817194797818275",
}
`;
